tests/jballerina-unit-test/src/test/resources/test-src/services/service_attach_test.bal

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+isolated boolean[] nameParamCheck = [false, false];
+
+public class Listener1 {
+    public isolated function 'start() returns error? { }
+
+    public isolated function gracefulStop() returns error? { }
+
+    public isolated function immediateStop() returns error? { }
+
+    public isolated function detach(service object {} s) returns error? { }
+
+    public isolated function attach(service object {} s, string? name) returns error? {
+        lock {
+            nameParamCheck[0] = name is () && !(<any> name is string[]);
+        }
+    }
+}
+
+service on new Listener1() {
+
+}
+
+public class Listener2 {
+    public isolated function 'start() returns error? { }
+
+    public isolated function gracefulStop() returns error? { }
+
+    public isolated function immediateStop() returns error? { }
+
+    public isolated function detach(service object {} s) returns error? { }
+
+    public isolated function attach(service object {} s, string? name) returns error? {
+        lock {
+            nameParamCheck[1] = name is string && name == "Service";
+        }
+    }
+}
+
+service "Service" on new Listener2() {
+
+}
+
+public function testAttachMethodParams() returns boolean {
+    lock {
+        return nameParamCheck[0] && nameParamCheck[1];
```

Comment:
Not directly related to this PR, but shall we add another service and test the scenario where the value is a string array if we don't have such a test already?

Reply:
@MaryamZi Does that work in normal case? To test the array case, we need to have a resource path right? Or should that test go to http module?

Reply:
It's when the service attach-point is an absolute-resource-path with resource-path-segments right?

https://ballerina.io/ballerina-spec/spec.html#attach-point

```ballerina
service /foo/bar on new Listener() {
    resource function get . () {
        
    }    
}
```